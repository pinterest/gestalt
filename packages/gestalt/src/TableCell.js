// @flow strict
import * as React from 'react';
import styles from './Table.css';

type Props = {|
  children: React.Node,
  colSpan?: number,
  rowSpan?: number,
|};

export default function TableCell(props: Props) {
  const { children, colSpan, rowSpan } = props;

  return (
    <td className={styles.td} colSpan={colSpan} rowSpan={rowSpan}>
      {children}
    </td>
  );
}
