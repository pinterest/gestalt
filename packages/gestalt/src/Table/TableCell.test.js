// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import TableCell from './TableCell.js';

test('renders correctly', () => {
  const tree = renderer.create(<TableCell>cell content</TableCell>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with colSpan and rowSpan', () => {
  const tree = renderer
    .create(
      <TableCell rowSpan={2} colSpan={3}>
        cell content
      </TableCell>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
