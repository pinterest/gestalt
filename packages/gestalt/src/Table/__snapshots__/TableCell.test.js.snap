// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<td
  className="td"
>
  cell content
</td>
`;

exports[`renders correctly with colSpan and rowSpan 1`] = `
<td
  className="td"
  colSpan={3}
  rowSpan={2}
>
  cell content
</td>
`;
